### ISSUE TITLE
Add gradient descent

### ISSUE THREAD

**Issue by RUBIKX107** [RUBIKX107/Spec-driven-development-#1]
--> instead of closed form slope formula
     --> iteratively improve slope + intercept
     --> this is exactly how neural networks train!
